doc/nyan/api_reference/reference_aux.md

```diff
@@ -1180,6 +1180,7 @@ A set of requirements that unlock the technology for the researching game entity
 Resource(Entity):
     name        : TranslatedString
     max_storage : int
+    autodrop    : bool
```

Comment:
This should not be in `Resource` since this is a global reference object. 

I would say that we already could do autodrop with existing abilities: If the game entity has both `Gather` and `DropSite` abilities for the same resource, then the resources are automatically added to the resource pool with the `gather_rate` rate. Alternatively, we could introduce a new `Modifier` that allows autodrop.

Reply:
> This should not be in Resource since this is a global reference object.

There is no problem being global.

> If the game entity has both Gather and DropSite

This is more complex and also for new resources introduced with nyan patches you cannot add a DropSite to every Gatherer

Reply:
>There is no problem being global.

It is a problem because the resource should not define if it is droppable. Rather, the villager object should define whether it is allowed to autodrop its resources.

> This is more complex and also for new resources introduced with nyan patches you cannot add a DropSite to every Gatherer

If you introduce a new resource, you can also patch `accepts` in `DropSite` to include it. It would be one additional patch. For complexity, it can be integrated in the search for drop sites that a game entity has to do anyway.

Reply:
You are right, I reliase now that there two cases

Case **1**: Favor (AoM) is autodrop, it does not depen on the villager.
Case **2**: VillagerWithACartForFood can autodrop Food, but Food is not autodrop.

So, I suggest we do both: There is a autodrop attribute to the resource and also we allow Gather with DropSite.

Agreed? (on yes, I will impl the special case of both Gather and DropSite)

Reply:
You should probably come to the [matrix channel](https://riot.im/app/#/room/#sfttech:matrix.org) for further discussion :) I've seen that you also made suggestion for nyan, but used outdated API ideas, so it would be good to get you up-to-date.

> There is a autodrop attribute to the resource and also we allow Gather with DropSite.

I still think that `autodrop` should not belong to `Resource`. Resources do not have to be gathered, they could be accumulated stored entities (relics), provided temporarily (pop space) or created out of thin air (cheats). The `Resource` object should therefore be neutral. Autodropping is better placed in the related gather abilities (`Gather` or `DropSite`), otherwise the member is redundant for a lot of cases.

Reply:
I agree that autodropping in general is not a property of a resource (type). The autodropping depends on where you want it do be dropped. I think any resource should be autodroppable, as long as there is an entity near enough that accept this resource as autodrop.

Reply:
@TheJJ @heinezen +1, that would also make some AoE3 behaviour possible if we raise the distance to something "infinite". So for example faith can be "gathered" due e.g. fighting all over the map with let's say having build the monastery as a `DropSite`.